src/lib.rs

```diff
@@ -274,6 +274,7 @@ pub fn make_test(config: &Config, testpaths: &TestPaths) -> test::TestDescAndFn
             name: make_test_name(config, testpaths),
             ignore: early_props.ignore,
             should_panic: should_panic,
+            allow_fail: false,
```

Comment:
This code is copied from the upstream `make_test()` function: https://github.com/rust-lang/rust/blob/919c4a6707da3aa2cc9ff63e33057e1e2a90720b/src/tools/compiletest/src/main.rs#L479.

I think what we really want to do is to parse the attributes of the test to determine the intended value of `allow_fail`, but that would be less trivial of a change. Right now I only care about getting compiletest to run again.

I don't see how it would make sense to use a field of `Config`, as that struct is shared among all tests.